code/game/machinery/washing_machine.dm

```diff
@@ -14,306 +13,401 @@
+	if(!istype(atom_to_insert) || !istype(user))
+		return FALSE
+	if(user.stat != CONSCIOUS || !Adjacent(user) || !atom_to_insert.Adjacent(user))
+		return FALSE
+	if(HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+		to_chat(user, "<span class='notice'>You try to insert [atom_to_insert] into [src] but your hands are blocked!</span>")
+		return FALSE
+	if(isliving(atom_to_insert)) // not ismob() b/c we only want living creatures, ghosts could just float out of the washing machine whenever they wanted
+		var/mob/living/sacrifice = atom_to_insert
+		if(sacrifice.mob_size >= MOB_SIZE_HUMAN) // prevents some most mobs from being washed even if it would be really really funny
+			to_chat(user, "<span class='notice'>You try to insert [sacrifice] into [src] but [sacrifice.p_theyre()] too large to fit inside the tub!</span>")
+			return FALSE
+		if(sacrifice.buckled)
+			to_chat(user, "<span class='notice'>You try to insert [sacrifice] into [src] but [sacrifice.p_theyre()] still buckled to something!</span>")
+			return FALSE
+		return insert_mob_into_tub(user, sacrifice)
+	if(isitem(atom_to_insert)) // Restricting this to only item types prevents a lot of issues, you really only wanna shove items in here anyway...
+		var/obj/item/item_to_insert = atom_to_insert
+		if(item_to_insert.anchored)
+			return FALSE
+		if(item_to_insert.w_class >= WEIGHT_CLASS_HUGE)
+			to_chat(user, "<span class='notice'>You try to insert [item_to_insert] into [src] but [item_to_insert.p_them()] is too large for the tub!</span>")
+			return FALSE
+		return insert_item_into_tub(user, item_to_insert)
+	return FALSE // how did we get here?
+
+/// Handles inserting mobs into the washing machines, checks machines capacity, does a do_after, and then applys appropriate signals and updates machines state
+/obj/machinery/washing_machine/proc/insert_mob_into_tub(mob/user, mob/living/mob_to_insert)
+	var/mob_content_size = mob_to_insert.mob_size
+	if(max_tub_capacity < ((mob_content_size * 2) + current_tub_capacity))
+		to_chat(user, "<span class='warning'>You try to insert [mob_to_insert] into [src] but the tub is too full for [mob_to_insert.p_them()]!</span>")
+		return FALSE
+	to_chat(mob_to_insert, "<span class='userdanger'>[user] starts shoving you into [src]!</span>")
+	visible_message("<span class='warning'>[user] starts shoving [mob_to_insert] into [src]!</span>")
+	if(!do_after_once(user, (3 SECONDS * mob_content_size), target = src, attempt_cancel_message = "You stop inserting [mob_to_insert] into [src]."))
+		return FALSE
+	mob_to_insert.forceMove(src)
+	RegisterSignal(mob_to_insert, COMSIG_PARENT_QDELETING, PROC_REF(check_tub_contents))
+	LAZYADD(inserted_mobs, mob_to_insert)
+	calculate_tub_capacity()
+	update_washing_state()
+	add_attack_logs(user, mob_to_insert, "Shoved into washing machine.")
+	to_chat(mob_to_insert, "<span class='userdanger'>[user] shoves you into [src]. Oh shit!</span>")
+	to_chat(user, "<span class='danger'>You shove [mob_to_insert] into [src].</span>")
```

Comment:
visible_message would probably be better here

Reply:
added in next commit